instrumentation/rxjava-2.0/library/src/main/java/io/opentelemetry/instrumentation/rxjava2/TracingConnectableObservable.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.rxjava2;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.reactivex.Observer;
+import io.reactivex.annotations.NonNull;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Consumer;
+import io.reactivex.observables.ConnectableObservable;
+
+class TracingConnectableObservable<T> extends ConnectableObservable<T> {
+
+  private final ConnectableObservable<T> source;
+  private final Context parentSpan;
+
+  TracingConnectableObservable(final ConnectableObservable<T> source, final Context parentSpan) {
+    this.source = source;
+    this.parentSpan = parentSpan;
+  }
+
+  @Override
+  public void connect(final @NonNull Consumer<? super Disposable> connection) {
```

Comment:
We don't use `NonNull`, instead we use `Nullable` where needed - can you remove `NonNull` throughout?

Reply:
sure, will do